### PR TITLE
chore(ci): add GitHub workflow to enforce Prettier check

### LINKED ISSUE #212: Prettier GitHub workflow
### Description

https://github.com/Orange-OpenSource/ods-charts/pull/198 introduced Prettier to the repository. Some npm commands can be run by the developers:
- `npm run prettier:update`
- `npm run prettier:check`

The idea of this issue would be to add a new GitHub workflow to check that every PR has run Prettier beforehand; run `npm run prettier:check`.

### PR DESCRIPTION
**Pull request by julien-deramond**
### Related issues

Closes #212

### Description

This PR adds a GitHub workflow that will enforce the Prettier formatting rules on all files in the repository:
- It will run on every push to the repository
- It will run on every pull request to the repository
- It can be run manually by anyone with write access to the repository

##### Testing

This PR came firstly with few lines added to `src/theme/common/ODS.line-style.broken.ts` in order to make the workflow fail voluntarily, as you can see at https://github.com/Orange-OpenSource/ods-charts/actions/runs/8811608553/job/24185800032?pr=214.

Confirming that everything works correctly, [this modification has been removed](https://github.com/Orange-OpenSource/ods-charts/pull/214/commits/03fe98be29f3e36b998d360bd1b9affe8e922ead), and the workflow is now green.

### Types of change

- New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+          cache: npm
+
+      - run: node --version
+      - run: npm --version
+
+      - name: Install npm dependencies
+        run: npm ci
+
+      - name: Check Prettier
+        run: npm run prettier:check
```
Comment (louismaximepiton) [inline .github/workflows/lint.yml:41]: This should run with `--max-warnings=0` option to send error on warning

**Review by louismaximepiton — COMMENTED** [review]
One point to discuss maybe

File: .github/workflows/lint.yml (hunk 1 above)
Comment (julien-deramond) [reply .github/workflows/lint.yml:41]: When it was run with https://github.com/Orange-OpenSource/ods-charts/actions/runs/8811608553/job/24185800032?pr=214 there were only warnings and the job failed. Sounds like it's the default behavior.

Comment (louismaximepiton) [reply .github/workflows/lint.yml:41]: My bad, I didn't saw this job

**Review by louismaximepiton — APPROVED** [review]

**Pull request merged**